statsmodels/multivariate/manova.py

```diff
@@ -0,0 +1,240 @@
+           (1 by k_exog, the number of independent variables) is required.
+           If an array of strings, it will be passed to
+           patsy.DesignInfo().linear_constraint.
+        transform_M : array-like or an array of strings
+            Transform matrix.
+            Default to be k_endog by k_endog identity matrix
+            (i.e. do not transform y matrix).
+           If an array of strings, it will be passed to
+           patsy.DesignInfo().linear_constraint.
+        If `hypothesis` is None: 1) the effect of each independent variable
+        on the dependent variables will be tested. Or 2) if model is
+        created using a formula,  `hypothesis` will be created according to
+        `design_info`. 1) and 2) is equivalent if no additional variables
+        are created by the formula (e.g. dummy variables for categorical
+        variables and interaction terms)
+    fittedvalues : tuple
+        Output of ``fit_manova``
+    exog_names : a list of string
+    endog_names : a list of string
+
+    Returns
+    -------
+    a dict of manova tables
+    """
+    def fn(L, M):
+        df_resid, u, sscpr = fittedvalues
+        t1 = L.dot(u)
+        q = matrix_rank(t1)
+        t1 = t1.dot(M)
+        H = t1.T.dot(t1)
+
+        # E = M'(Y'Y - B'(X'X)B)M
+        E = M.T.dot(sscpr).dot(M)
+        return E, H, q, df_resid
+
+    return _multivariate_htest(hypothesis, exog_names, endog_names, fn)
+
+
+class MANOVA(Model):
+    """
+    Multivariate analysis of variance
+
+
+    Parameters
+    ----------
+    endog : array-like
+        Dependent variables (DV). A n_sample x n_y_var array where n_sample is
+        the number of observations and n_y_var is the number of DV.
+
+    exog : array-like
+        Independent variables (IV). A n_sample x n_x_var array where n is the
+        number of observations and n_x_var is the number of IV. An intercept is
+        not included by default and should be added by the user (models
+        specified using a formula include an intercept by default)
+
+    .. [1] ftp://public.dhe.ibm.com/software/analytics/spss/documentation/statistics/20.0/en/client/Manuals/IBM_SPSS_Statistics_Algorithms.pdf
+
+    Attributes
+    -----------
+    df_resid : float
+        The number of observation `n` minus the number of IV `q`.
+    sscpr : array
+        Sums of squares and cross-products of residuals
+    endog : array
+        See Parameters.
+    exog : array
+        See Parameters.
+    design_info : patsy.DesignInfo
+        Contain design info for the independent variables if model is
+        constructed using `from_formula`
+
+    """
+    def __init__(self, endog, exog, design_info=None, **kwargs):
+        self.design_info = design_info
+        super(MANOVA, self).__init__(endog, exog)
+        self.fittedmod = manova_fit(self.exog, self.endog)
+
+    @classmethod
+    def from_formula(cls, formula, data, subset=None, drop_cols=None,
```

Comment:
this looks redundant, Isn't it all inherited

There are some things that I don't understand well enough in the formula interface and associated data handling.
e.g. I don't see how the design_info as `__init__` argument works.